3rd-party/cove/app/Thread.cpp

```diff
@@ -21,29 +21,27 @@
 
 #include "libvectorizer/Vectorizer.h"
 
-#include "UIProgressDialog.h"
-
 class QObject;
 
 namespace cove {
 //@{
 //! \ingroup gui
 
 /*! \class ClassificationThread
-  \brief Subclass of Thread, used for running classification process. */
-/*! \var UIProgressDialog* ClassificationThread::p
-  UIProgressDialog to be passed to performClassification. Can be 0. */
+  \brief Subclass of QThread, used for running classification process. */
 /*! \var Vectorizer* ClassificationThread::v
   Vectorizer on which to run performClassification. */
+/*! \var ProgressObserver* ClassificationThread::p
+  ProgressObserver, e.g. UIProgressDialog, to be passed to performClassification. */
 
 /*! Constructor.
   \param[in] v Vectorizer on which to run performClassification.
   \param[in] p UIProgressDialog to be passed to performClassification. Can be 0.
   */
-ClassificationThread::ClassificationThread(Vectorizer& v, UIProgressDialog* const p, QObject* parent)
-    : v(v)
+ClassificationThread::ClassificationThread(Vectorizer& v, ProgressObserver* const p, QObject* parent)
+    : QThread(parent)
+    , v(v)
     , p(p)
-    , QThread(parent)
```

Comment:
I've stopped here for a moment and yes, this is completely correct. This code was written long before Guidelines came into existence. Hence the "wrong" initialization order. Thanks for the fixes!